WordPress/Classes/ViewRelated/NewGutenberg/NewGutenbergViewController.swift

```diff
@@ -891,6 +840,73 @@ extension NewGutenbergViewController {
     }
 }
 
+extension EditorConfiguration {
+    init(blog: Blog) {
+        let selfHostedApiUrl = blog.url(withPath: "wp-json/")
+        let isWPComSite = blog.isHostedAtWPcom || blog.isAtomic()
+        let siteApiRoot = blog.isAccessibleThroughWPCom() && isWPComSite ? blog.wordPressComRestApi?.baseURL.absoluteString : selfHostedApiUrl
+        let siteId = blog.dotComID?.stringValue
+        let siteDomain = blog.primaryDomainAddress
+        let authToken = blog.authToken ?? ""
+        var authHeader = "Bearer \(authToken)"
+
+        let applicationPassword = try? blog.getApplicationToken()
+
+        if let appPassword = applicationPassword, let username = blog.username {
+            let credentials = "\(username):\(appPassword)"
+            if let credentialsData = credentials.data(using: .utf8) {
+                let base64Credentials = credentialsData.base64EncodedString()
+                authHeader = "Basic \(base64Credentials)"
+            }
+        }
+
+        // Must provide both namespace forms to detect usages of both forms in third-party code
+        var siteApiNamespace: [String] = []
+        if isWPComSite {
+            if let siteId {
+                siteApiNamespace.append("sites/\(siteId)")
+            } else {
+                siteApiNamespace.append("sites/\(siteDomain)")
+            }
+        }
+
+        self = EditorConfiguration()
+
+        self.siteURL = blog.url ?? ""
+        self.siteApiRoot = siteApiRoot ?? ""
+        self.siteApiNamespace = siteApiNamespace
+        self.namespaceExcludedPaths = ["/wpcom/v2/following/recommendations", "/wpcom/v2/following/mine"]
+        self.authHeader = authHeader
+
+        self.themeStyles = FeatureFlag.newGutenbergThemeStyles.enabled
+        // Limited to Simple sites until application password auth is supported
+        if RemoteFeatureFlag.newGutenbergPlugins.enabled() && blog.isHostedAtWPcom {
+            self.plugins = true
+            if var editorAssetsEndpoint = blog.wordPressComRestApi?.baseURL {
+                editorAssetsEndpoint.appendPathComponent("wpcom/v2")
+                for part in siteApiNamespace {
+                    editorAssetsEndpoint.appendPathComponent(part)
+                }
+                editorAssetsEndpoint.appendPathComponent("editor-assets")
+                self.editorAssetsEndpoint = editorAssetsEndpoint
+            }
+        }
```

Comment:
This if statement is new. The rest are moved from the view controller initializer above.